examples/placegoose/src/routes/gaggles.ts

```diff
@@ -0,0 +1,125 @@
+import { eq } from "drizzle-orm";
+import { drizzle } from "drizzle-orm/d1";
+import { Hono } from "hono";
+import { HTTPException } from "hono/http-exception";
+import { validator } from "hono/validator";
+import { z } from "zod";
+
+import { getGaggleById } from "../controllers";
+import * as schema from "../db/schema";
+import { generateId } from "../lib/utils";
+import { makeBodyValidator, validateIdParam } from "../lib/validation";
+import type { DatabaseBindings } from "../types";
+
+const ZGaggleInsert = z.object({
+  name: z.string().min(1),
+  territory: z.string().min(1).nullable(),
+});
+
+const gagglesApp = new Hono<{ Bindings: DatabaseBindings }>();
+
+// Get all Gaggles
+gagglesApp.get("/", async (c) => {
+  const db = drizzle(c.env.DB);
+  const gaggles = await db.select().from(schema.gaggles);
+
+  return c.json(gaggles);
+});
+
+// Create a new Gaggle
+gagglesApp.post(
+  "/",
+  validator("json", makeBodyValidator(ZGaggleInsert.parse)),
```

Comment:
not sure if you evaluated it already, but there is a specific package for doing valdiation with zod schemas in hono! 

`@hono/zod-validator` see: https://hono.dev/docs/guides/validation#zod-validator-middleware

if you elected not to use that for a specific reason, could be worthwhile to explain that somewhere in the codebase

Reply:
right on. i think we could do a section on this in the write-up, since it'd be good to call out existing solutions ((esp mentioning that it'd still requrie a custom error handler))